java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -63,11 +64,20 @@ public boolean equals(Object o) {
       return false;
     }
     Dictionary that = (Dictionary) o;
-    return Objects.equals(encoding, that.encoding) && Objects.equals(dictionary, that.dictionary);
+    return Objects.equals(encoding, that.encoding) && compareFieldVector(dictionary, that.dictionary);
   }
 
   @Override
   public int hashCode() {
     return Objects.hash(encoding, dictionary);
   }
+
+  private boolean compareFieldVector(FieldVector vector1, FieldVector vector2) {
```

Comment:
Could you add a TODO note here to clean this up once we can compare vectors without using the Validator?

Reply:
Thanks, fixed.